staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go

```diff
@@ -139,15 +140,15 @@ func (f *TestFieldManager) Reset() {
 }
 
 func (f *TestFieldManager) Apply(obj runtime.Object, manager string, force bool) error {
-	out, err := fieldmanager.NewFieldManager(f.fieldManager).Apply(f.liveObj, obj, manager, force)
+	out, err := fieldmanager.NewFieldManager(f.fieldManager, f.isSubresource).Apply(f.liveObj, obj, manager, force)
```

Comment:
I am totally fine to have a new struct `TestSubresourceFieldManager` which embeds a `TestFieldManager`. However the method above become something like:

```go
func (f *TestFieldManager) Apply(obj runtime.Object, manager string, force bool) error {
    out, err := fieldmanager.NewFieldManager(f.fieldManager, false).Apply(f.liveObj, obj, manager, force)
    // ....
}

I am not sure the hardcoded `false` feels better. I am not sure if I missed your point though.

Reply:
@apelisse which PR are we waiting for?

Reply:
Sorry, that's kind of unfortunate. @julianvmodesto knows, he has multiple I'm not sure which it is. @nodo you could then probably rebase your change on top of his so that you can start making progress.

Reply:
Ah this one: https://github.com/kubernetes/kubernetes/pull/84134

I think it is close to merging.

See the field_manager_test.go, where we use an entire FieldManager in the test now.